src/autolabel/tasks/classification.py

```diff
@@ -55,6 +60,40 @@ def construct_prompt(self, input: Dict, examples: List) -> str:
         # prepare task guideline
         labels_list = self.config.labels_list()
         num_labels = len(labels_list)
+
+        # if large number of labels, filter labels_list by similarity of labels to input
+        if num_labels >= 50:
+            example_prompt = PromptTemplate(
+                input_variables=["input"],
+                template="{input}",
+            )
+            label_examples = [{"input": label} for label in labels_list]
+
+            example_selector = SemanticSimilarityExampleSelector.from_examples(
+                # This is the list of labels available to select from.
+                label_examples,
+                # This is the embedding class used to produce embeddings which are used to measure semantic similarity.
+                OpenAIEmbeddings(),
```

Comment:
can we use the embedding model that is the same as the one used for the seed examples, this can be read from the config

Reply:
+1

Reply:
done in latest commit.

It now chooses embedding function based on `config.embedding_provider()`
```
            self.label_selector = LabelSelector.from_examples(
                labels=self.config.labels_list(),
                k=self.config.label_selection_count(),
                embedding_func=PROVIDER_TO_MODEL.get(
                    self.config.embedding_provider(), DEFAULT_EMBEDDING_PROVIDER
                )(),
            )
```